src/phoenix/trace/llama_index/callback.py

```diff
@@ -437,10 +449,18 @@ def _message_payload_to_attributes(message: Any) -> Dict[str, Optional[str]]:
         # NB: these additional kwargs exist both for 'agent' and 'function' roles
         if "name" in message.additional_kwargs:
             message_attributes[MESSAGE_NAME] = message.additional_kwargs["name"]
-        if "function_call" in message.additional_kwargs:
-            function_call = message.additional_kwargs["function_call"]
-            message_attributes[MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON] = function_call.arguments
-            message_attributes[MESSAGE_FUNCTION_CALL_NAME] = function_call.name
+        if tool_calls := message.additional_kwargs.get("tool_calls"):
+            assert isinstance(
+                tool_calls, Iterable
+            ), f"tool_calls must be Iterable, found {type(tool_calls)}"
+            tool_call = next(iter(tool_calls), None)
+            function = getattr(tool_call, "function", None)
+            if name := getattr(function, "name", None):
+                assert isinstance(name, str), f"name must be str, found {type(name)}"
+                message_attributes[MESSAGE_FUNCTION_CALL_NAME] = name
+            if arguments := getattr(function, "arguments", None):
+                assert isinstance(arguments, str), f"arguments must be str, found {type(arguments)}"
+                message_attributes[MESSAGE_FUNCTION_CALL_ARGUMENTS_JSON] = arguments
```

Comment:
is this related to the multi-function calling in GPT-4-Turbo? If so we probably need a follow-up ticket. Seems like we are only parsing the first tool call?

Reply:
yea, it complicates everything. i will add a follow-up ticket